### PR TITLE
add plugin weblogic_cve_2020_14882

### PR DESCRIPTION
**Pull request by thiscodecc**
Hey,

this PR for the Vuln Detector Plugin for CVE-2020-14882 (see Issue #25  )

In order to verify the vulnerability, I used two different docker environments, one for the vulnerable version and the other for the fixed version

The first step is to clone my vulnerability environment warehouse to the local

Vulnerable image:
```
docker pull thisvul/weblogic14cjdk8u
docker run -d -p 7001:7001 --name weblogic14cjdk8u thisvul/weblogic14cjdk8u
```

Safe image:
```
docker pull thisvul/weblogic_not_14882
docker run -d -p 7001:7001 --name weblogic_not_14882 thisvul/weblogic_not_14882
```

### CODE REVIEW

File: community/detectors/weblogic_cve_2020_14882/build.gradle [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    implementation "com.google.tsunami:tsunami-proto:${tsunamiVersion}"
+
+    testImplementation "junit:junit:${junitVersion}"
+    testImplementation "org.mockito:mockito-core:${mockitoVersion}"
+    testImplementation "com.google.truth:truth:${truthVersion}"
+    testImplementation "com.google.truth.extensions:truth-java8-extension:${truthVersion}"
+    testImplementation "com.google.truth.extensions:truth-proto-extension:${truthVersion}"
+
+    testImplementation "com.squareup.okhttp3:mockwebserver:${okhttpVersion}"
+}
```
Comment (magl0) [inline community/detectors/weblogic_cve_2020_14882/build.gradle:69]: Code style: can you please add a terminating newline to the end of this file?

File: community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882DetectorBootstrapModule.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (magl0) [inline community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882DetectorBootstrapModule.java:2]: Update this to `2021` please.

File: community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetector.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
+package com.google.tsunami.plugins.detectors.cves.cve202014882;
```
Comment (magl0) [inline community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetector.java:1]: Please add the license header to the top of this file.

File: community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetector.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + CHECK_VUL_PATH;
+
+    try {
+      HttpResponse httpResponse =
+          httpClient.send(get(targetUri).withEmptyHeaders().build(), networkService);
+
+      //Solve the WebLogic http 302 redirection problem
```
Comment (magl0) [inline community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetector.java:91]: Code style: please add a space after `//`.

File: community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetector.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+      Optional<String> cookie = httpResponse.headers().get("Set-Cookie");
```
Comment (magl0) [inline community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetector.java:92]: Please use `com.google.common.net.HttpHeaders.SET_COOKIE`, similar to the `COOKIE` header on line 97.

File: community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetector.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+            httpClient.send(get(targetUri).withEmptyHeaders()
+                    .setHeaders(HttpHeaders.builder().addHeader(COOKIE, cookie.get()).build()).build(),
+                networkService);
+      }
+
+      if (httpResponse.status().code() == 200
+          && httpResponse.bodyString().get().contains(DETECTION_STRING)) {
+        return true;
+      }
+
```
Comment (magl0) [inline community/detectors/weblogic_cve_2020_14882/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetector.java:105]: Code style: remove this empty line.

File: community/detectors/weblogic_cve_2020_14882/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetectorTest.java [hunk 7]
```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (magl0) [inline community/detectors/weblogic_cve_2020_14882/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetectorTest.java:2]: Same, `2021`.

File: community/detectors/weblogic_cve_2020_14882/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetectorTest.java [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    mockWebServer.shutdown();
+  }
+
+  @Test
+  public void detect_whenVulnerable_returnsVulnerability() throws IOException {
+
```
Comment (magl0) [inline community/detectors/weblogic_cve_2020_14882/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetectorTest.java:79]: Code style: remove this empty line.

File: community/detectors/weblogic_cve_2020_14882/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetectorTest.java [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+            .build();
+
+    DetectionReportList detectionReports = detector.detect(targetInfo, httpServices);
+
+    assertThat(detectionReports.getDetectionReportsList()).isEmpty();
+  }
+
+  private void mockWebResponse(String body) throws IOException {
+    mockWebServer.start();
+    mockWebServer.enqueue(new MockResponse().setResponseCode(200).setBody(body));
```
Comment (magl0) [inline community/detectors/weblogic_cve_2020_14882/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetectorTest.java:147]: The detector has a logic to handle `302` redirect with the `Set-Cookie` header. Can you please add a unit test for this case as well?

**Comment by thiscodecc** [comment]
I resubmitted the code again, waiting for your review

**Review by magl0 — COMMENTED** [review]
This looks OK. This PR will be redirected to an internal reviewer for code reviews and I'll kick-off the payload verification process. Feedbacks will be mirrored back to this PR.

**Comment by magl0** [comment]
Hi @thiscodecc, can you please run [google-java-format](https://github.com/google/google-java-format) against this PR to reformat all the source code?

**Comment by thiscodecc** [comment]
image

I have used google-java-format to format this project. Are there any questions?

**Comment by magl0** [comment]
Did you use the google-java-format Intellij plugin by following [this instruction](https://github.com/google/google-java-format#intellij-android-studio-and-other-jetbrains-ides)? Can you instead try to execute the tool via command line by following [this instruction](https://github.com/google/google-java-format#from-the-command-line)?

**Comment by thiscodecc** [comment]
I just executed google-java-format using the command line, please check it out.

**Comment by magl0** [comment]
Thanks! The internal linter is happy now.

**Comment by thiscodecc** [comment]
This is done , Can I start another issue? #207

**Pull request merged**